sdk/keyvault/keyvault-secrets/src/index.ts

```diff
@@ -809,20 +831,29 @@ export class SecretsClient {
   /**
    * Creates a span using the tracer that was set by the user
    * @param methodName The name of the method for which the span is being created.
-   * @param requestOptions The options for the underlying http request. This will be
-   * updated to use the newly created span as the "parent" so that any new spans created
-   * after this point gets the right parent.
+   * @param requestOptions The options for the underlying http request.
    */
   private createSpan(methodName: string, requestOptions?: RequestOptionsBase): Span {
     const tracer = TracerProxy.getTracer();
-    const options = requestOptions || {};
-    const span = tracer.startSpan(methodName, options.spanOptions);
-    if (
-      tracer.pluginType !== SupportedPlugins.NOOP &&
-      (options.spanOptions && options.spanOptions.parent)
-    ) {
-      options.spanOptions = { ...options.spanOptions, parent: span };
+    return tracer.startSpan(methodName, requestOptions && requestOptions.spanOptions);
+  }
+
+  /**
+   * Updates HTTP options to include the given span as the parent of future spans
+   * @param span The span for the current operation
+   * @param options The options for the underlying http request
+   */
+  private setParentSpan(span: Span, options: RequestOptionsBase = {}): RequestOptionsBase {
+    if (span.isRecordingEvents()) {
```

Comment:
This `if` ensures that we dont update `options` when dealing with no-op spans/tracers
What about the scenario where user has passed no parent?
I am not sure what the spec says regarding this scenario, but I have the vague recollection that we do not want create spans if user has not explicitly created one and passed it as a "parent" to us.

@chradek Do you remember anything regarding this? If not, can you check with other folks working on tracing and get back here?

Reply:
If I understand the scenario, are you saying that if the user doesn't specify a parent span explicitly but does set a real Tracer (and not our no-op implementation) we wouldn't want to log any spans? 

I would think that if the user gives a real Tracer, then they would expect our tracing to be enabled, even if they don't manually pass us anything, but maybe I'm confused here.

Reply:
We should still be getting the same behavior here. If the user hasn't passed us a parent span, then `createSpan` will create a no-op span since there won't be a traceId to associate with it. In that case, this if statement won't set the parent span either.